Meta/CMake/Superbuild/CMakeLists.txt

```
@@ -0,0 +1,79 @@
+cmake_minimum_required(VERSION 3.16)
+
+# Variables that we really don't want the user to pass to the serenity cross-build because they will screw things up :^)
+set(LAGOM_ONLY_VARS "CMAKE_CXX_COMPILER;CMAKE_C_COMPILER")
+set(EXCLUDE_FROM_CL_VARS "SERENITY_ARCH;SERENITY_TOOLCHAIN")
+
+# https://stackoverflow.com/a/48555098
+# https://cmake.org/pipermail/cmake/2018-January/067002.html
+# MUST be done before call to 'project'
+get_cmake_property(vars CACHE_VARIABLES)
+foreach(var IN LISTS vars)
+    get_property(currentHelpString CACHE "${var}" PROPERTY HELPSTRING)
+    if("${currentHelpString}" MATCHES "No help, variable specified on the command line." OR "${currentHelpString}" STREQUAL "")
+        message("${var} = [${${var}}]  --  ${currentHelpString}") # uncomment to see the variables being processed
+        if (var IN_LIST EXCLUDE_FROM_CL_VARS)
+            continue()
+        elseif(var IN_LIST LAGOM_ONLY_VARS)
+            list(APPEND LAGOM_CL_ARGS "-D${var}=${${var}}")
+        else()
+            list(APPEND CL_ARGS "-D${var}=${${var}}")
+        endif()
+    endif()
+endforeach()
+
+list(APPEND LAGOM_CL_ARGS ${CL_ARGS})
+
+project(
+    SerenitySuperbuild
+    DESCRIPTION "Orchestrate host and target builds in a single build"
+    LANGUAGES NONE
+)
+
+get_filename_component(
+    SERENITY_SOURCE_DIR "${PROJECT_SOURCE_DIR}/../../.."
+    ABSOLUTE CACHE
+)
+set(SERENITY_ARCH "i686" CACHE STRING "Architecture to build for (i686 or x86_64)")
+set(SERENITY_TOOLCHAIN "GNU" CACHE STRING "Compliler toolchain to use for Serenity (GNU or Clang)")
+set(SERENITY_BUILD_DIR "${PROJECT_BINARY_DIR}/../${SERENITY_ARCH}")
+
+include(ExternalProject)
+
+ExternalProject_Add(
+    lagom
+    SOURCE_DIR "${SERENITY_SOURCE_DIR}/Meta/Lagom"
+    BINARY_DIR "${PROJECT_BINARY_DIR}/../lagom"
+    INSTALL_DIR "${PROJECT_BINARY_DIR}/../lagom-install"
+    CMAKE_GENERATOR Ninja
+    EXCLUDE_FROM_ALL YES
+    CMAKE_ARGS ${LAGOM_CL_ARGS}
+    CMAKE_CACHE_DEFAULT_ARGS
+    "-DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>"
+    "-DUNICODE_CACHE_LOCATION:PATH=${SERENITY_BUILD_DIR}"
+    "-DBUILD_LAGOM:BOOL=${BUILD_LAGOM}"
```

[Docs](https://cmake.org/cmake/help/latest/module/ExternalProject.html):

> The arguments are expected to be in the form `-Dvar:STRING=value`, which are then transformed into CMake `set()` commands ...

```suggestion
    -DCMAKE_INSTALL_PREFIX:STRING=<INSTALL_DIR>
    "-DUNICODE_CACHE_LOCATION:STRING=${SERENITY_BUILD_DIR}"
    "-DBUILD_LAGOM:STRING=${BUILD_LAGOM}"
```

This pattern is used by EP only to ensure that list values are properly processed.

Same below as well.